moped-database/migrations/1699408061246_alter_table_public_moped_users_add_column_last_seen_date/up.sql

```diff
@@ -0,0 +1,38 @@
+ALTER TABLE "public"."moped_users"
+    ADD COLUMN "last_seen_date" timestamptz NULL;
+
+COMMENT ON COLUMN moped_users.last_seen_date IS 'Tracks the last time a user loaded the Moped app in their browser. This value is set by the set_last_seen_date function. This value is not 100% reliable because it is updated by an API call that can be blocked by the client.';
+
+-- custom function which can be tracked as a hasura mutation
+-- https://hasura.io/docs/latest/schema/postgres/custom-functions/
+
+CREATE FUNCTION set_last_seen_date (hasura_session json)
+    RETURNS SETOF moped_users
```

Comment:
i tried a few different ways to condense this into fewer SQL operations, but it's really hard to debug.  Hasura requires that it returns a `moped_users` object, and the graphql endpoint takes the shape of that table.

```graphql
mutation SetLastSeenDate {
  set_last_seen_date {
    user_id
    first_name
    last_name
    is_user_group_member
    last_seen_date
  }
}
```

Reply:
super cool to see how this is set up when I checked it out in the Hasura console. Thanks for the link too - [accessing session data](https://hasura.io/docs/latest/schema/postgres/custom-functions/#accessing-hasura-session-variables-in-custom-functions) this way is a big TIL for me.